consensus/istanbul/backend/api.go

```diff
@@ -153,34 +181,15 @@ var (
 
 // GetCouncil retrieves the list of authorized validators at the specified block.
 func (api *APIExtension) GetCouncil(number *rpc.BlockNumber) ([]common.Address, error) {
-	// Retrieve the requested block number (or current if none requested)
-	var header *types.Header
-	if number == nil || *number == rpc.LatestBlockNumber {
-		header = api.chain.CurrentHeader()
-	} else if *number == rpc.PendingBlockNumber {
-		logger.Trace("Cannot get council of the pending block.", "number", number)
-		return nil, errPendingNotAllowed
-	} else {
-		header = api.chain.GetHeaderByNumber(uint64(number.Int64()))
-	}
-	// Ensure we have an actually valid block and return the council from its snapshot
-	if header == nil {
-		return nil, errNoBlockExist // return nil if block is not found.
+	previousBlockNumber := rpc.BlockNumber(number.Int64() - 1)
```

Comment:
Does it work with LatestBlockNumber and PendingBlockNumber?

Reply:
No. I should change the logic.
Thanks for your review!

Reply:
I reflected in https://github.com/klaytn/klaytn/pull/1468/commits/ac760f0f64b9ef6a18c45efc4edd83299c8a1eb7.
Thank you.

Reply:
I still not convinced how this code works with LatestBlockNumber.

If you see [networks/rpc/types.go](https://github.com/klaytn/klaytn/blob/dev/networks/rpc/types.go#L135), LatestBlockNumber == -1 and PendingBlockNumber == -2.
If `number == Latest`, then `rpc.BlockNumber(number.Int64() - 1)` is `-2` (Pending), not `head - 1` (What you might intended).

How about factoring out rpc.Number parsing using something like this?

```go
// a private function, not a method of API nor APIExtension.

// Retrieve the header at requested block number
func headerByRpcNumber(number *rpc.Number) (*types.Header, error) {
	var header *types.Header
	if number == nil || *number == rpc.LatestBlockNumber {
		header = api.chain.CurrentHeader()
	} else if *number == rpc.PendingBlockNumber {
		logger.Trace("Cannot get snapshot of the pending block.", "number", number)
		return nil, errPendingNotAllowed
	} else {
		header = api.chain.GetHeaderByNumber(uint64(number.Int64()))
	}
	// Ensure we have an actually valid block and return its snapshot
	if header == nil {
		return nil, errUnknownBlock // return nil if block is not found.
	}
	return header, nil
}
```

Then use like

```go
func (api *APIExtension) GetCouncil(number *rpc.BlockNumber) ([]common.Address, error) {
    header, err := headerByRpcNumber(number)
    if err != nil {
        return nil, err
    }
    
    snap, err := api.GetSnapshotAtHash(header.ParentHash)
    if err != nil {
        logger.Error("Failed to get snapshot.", "hash", snap.Hash, "err", err)
        return nil, errInternalError
    }
    return append(snap.validators(), snap.demotedValidators()...), nil
}
```

Reply:
Reflected in https://github.com/klaytn/klaytn/pull/1468/commits/8b3142b26d372e9e286c0a5495e53a1d4a7915b0.
Thank you!

Reply:
Reflected in https://github.com/klaytn/klaytn/pull/1468/commits/307820c288c4d7d1871a57d4b03f849deb46fdd0.
Thank you!